anxcloud/resource_dns_zone.go

```diff
@@ -0,0 +1,177 @@
+package anxcloud
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"go.anx.io/go-anxcloud/pkg/api"
+	clouddnsv1 "go.anx.io/go-anxcloud/pkg/apis/clouddns/v1"
+)
+
+func resourceDNSZone() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceDNSZoneCreate,
+		ReadContext:   resourceDNSZoneRead,
+		UpdateContext: resourceDNSZoneUpdate,
+		DeleteContext: resourceDNSZoneDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(1 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+		Schema: schemaDNSZone(),
+	}
+}
+
+func resourceDNSZoneCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	a := m.(providerContext).api
+
+	// try to import
+	z := &clouddnsv1.Zone{Name: d.Get("name").(string)}
```

Comment:
common usage of generic client is `a.Get(ctx, &object)`, so please have `z` not be a pointer and use `a.Get(ctx, &z)` below